lib/go-tc/servers.go

```diff
@@ -39,36 +42,36 @@ type Server struct {
 	DomainName       string              `json:"domainName" db:"domain_name"`
 	FQDN             *string             `json:"fqdn,omitempty"`
 	FqdnTime         time.Time           `json:"-"`
-	GUID             string              `json:"guid" db:"guid"`
+	GUID             *string             `json:"guid" db:"guid"`
```

Comment:
These fields are used for decoding, not just encoding. Are we 100% certain nil values in them won't cause nil access panics in apps that use these objects returned by the Client, like Traffic Monitor and Traffic Stats?

Even if we are, this could break anyone who happens to be using the `trafficops/client`, and has objects turn into pointers underneath them, and give them nil access panics at runtime.

I'm not saying it's not the right thing to do ideologically. But it's very dangerous, and makes me very nervous about Traffic Monitor.

Reply:
yes -- I'm aware of that,  and appreciate your input..   I think we should hash out these before this is merged.

It's a tough balancing act ensuring compatibility with the existing perl api.  My hope is that we can get these discrepancies ironed out sooner rather than later...